### PR TITLE
Adds admin list functionality

### PR DESCRIPTION
**Pull request by cameronvoell**
Part of https://github.com/xmtp/libxmtp/issues/664

PR adds the following

- isAdmin / isSuperAdmin
- adminList() \ superAdminList()
- add / remove admin
- updates group creation options to AllMembers / AdminOnly
- renames functions to isCreator / creatorAddress()

~~This PR is waiting on merging libxmtp changes into `identity-release` branch~~

### CODE REVIEW

File: library/src/main/java/org/xmtp/android/library/Group.kt [hunk 1]
```diff
@@ -215,6 +220,38 @@ class Group(val client: Client, private val libXMTPGroup: FfiGroup) {
         return libXMTPGroup.updateGroupName(name)
     }
 
+    fun isAdmin(inboxId: String): Boolean {
+        return libXMTPGroup.isAdmin(inboxId)
+    }
+
+    fun isSuperAdmin(inboxId: String): Boolean {
+        return libXMTPGroup.isSuperAdmin(inboxId)
+    }
```
Comment (nplasterer) [inline library/src/main/java/org/xmtp/android/library/Group.kt:229]: Can we have another function on member that tells the permission level for the member? https://github.com/xmtp/xmtp-android/blob/main/library/src/main/java/org/xmtp/android/library/libxmtp/Member.kt

File: library/src/main/java/org/xmtp/android/library/Group.kt [hunk 2]
```diff
@@ -215,6 +220,38 @@ class Group(val client: Client, private val libXMTPGroup: FfiGroup) {
...
+
+    suspend fun addAdmin(inboxId: String) {
+        return libXMTPGroup.addAdmin(inboxId)
```
Comment (nplasterer) [inline library/src/main/java/org/xmtp/android/library/Group.kt:232]: Do these error if they don't have permission to add an admin? Wondering if we should catch and surface a more friendly XMTPException?

**Review by nplasterer — APPROVED** [review]
Looks great just minor questions.

File: library/src/main/java/org/xmtp/android/library/Group.kt (hunk 1 above)
Comment (cameronvoell) [reply library/src/main/java/org/xmtp/android/library/Group.kt:229]: like this idea, add here dab330a326c523b378191d2ac7656563eb1f7d16

File: library/src/main/java/org/xmtp/android/library/Group.kt (hunk 2 above)
Comment (cameronvoell) [reply library/src/main/java/org/xmtp/android/library/Group.kt]: Added error handling here and made an issue to remind myself to improve the error messaging in libxmtp so we can surface more specific errors - 115b370bf624440cf8e4664597b5524a2f033471

File: library/src/main/java/org/xmtp/android/library/libxmtp/Member.kt [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
 import uniffi.xmtpv3.FfiGroupMember
+import uniffi.xmtpv3.FfiPermissionLevel
 
 class Member(private val ffiMember: FfiGroupMember) {
-
     val inboxId: String
         get() = ffiMember.inboxId
     val addresses: List<String>
         get() = ffiMember.accountAddresses
+    val permissionLevel: FfiPermissionLevel
```
Comment (nplasterer) [inline library/src/main/java/org/xmtp/android/library/libxmtp/Member.kt:11]: I like to typecast the ffi stuff to a friendly name. Kind of like what I did here: https://github.com/xmtp/xmtp-android/blob/main/library/src/main/java/org/xmtp/android/library/libxmtp/MessageV3.kt#L32-L37

Comment (cameronvoell) [reply library/src/main/java/org/xmtp/android/library/libxmtp/Member.kt]: updated here https://github.com/xmtp/xmtp-android/pull/245/commits/4fde146ac2d91bac0faee2a5597e59573d90b46c

**Pull request merged**